build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/HeadlessBuilder.java

```diff
@@ -1030,4 +1042,8 @@ protected void resetToolOptions(IConfiguration configuration) throws BuildExcept
 	@Override
 	public void stop() {
 	}
+
+	private static boolean isSpecialDirectory(String directoryName) {
+		return directoryName.equals(".metadata") || directoryName.equals(".org.eclipse.egit.core.cmp"); //$NON-NLS-1$ //$NON-NLS-2$
+	}
```

Comment:
Use a set to make it a bit easier to add new entries later? (Untested code)

```suggestion
	private static Set<String> specials = Set.of(".metadata", ".org.eclipse.egit.core.cmp"); //$NON-NLS-1$ //$NON-NLS-2$
	private static boolean isSpecialDirectory(String directoryName) {
		return specials.contains(directoryName);
	}
```

Reply:
You can probably then inline isSpecialDirectory too if you accept my suggested change.